# Project Cycle 2 - Group 4 - 03 inference
## 1. Load Processed Data

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

sad_df = pd.read_csv("../data/processed/sad_cleaned.csv")
bmi_df = pd.read_csv("../data/processed/bmi_cleaned.csv")
joint_df = pd.read_csv("../data/processed/joint_cleaned.csv")

## 2. Proportion Analysis (SadOrHopeless)
### 2-1 Sample Proportion

In [2]:
p_hat = sad_df["Sad_binary"].mean()
n = len(sad_df)

p_hat, n

(0.29996388587937883, 13845)

The sample proportion of students who feel sad or hopeless is approximately 0.30 (p̂ ≈ 0.29996) based on a sample size of 13,845. This estimate is very close to the benchmark value (p0 = 0.30), suggesting that there may not be a substantial difference.

### 2-2 Confidence Interval for Proportion

In [3]:
z = 1.96
se = np.sqrt(p_hat * (1 - p_hat) / n)

ci_lower = p_hat - z * se
ci_upper = p_hat + z * se

(ci_lower, ci_upper)

(0.2923307270442763, 0.30759704471448135)

The 95% confidence interval for the population proportion is approximately (0.2923, 0.3076). Since the benchmark value (p0 = 0.30) lies within this interval, there is no strong evidence that the true proportion differs from 0.30.

### 2-3 One-Sample Proportion Test

In [4]:
p0 = 0.30

z_stat = (p_hat - p0) / se
p_value = 2 * (1 - stats.norm.cdf(abs(z_stat)))

z_stat, p_value

(-0.009273182694948381, 0.9926011767381813)

The p-value is approximately 0.9926, which is much greater than the significance level (α = 0.05). Therefore, we fail to reject the null hypothesis. There is no sufficient evidence to conclude that the true proportion of students who feel sad or hopeless is different from 0.30.

## 3. Mean Analysis of BMIPCT
### 3-1 Sample Mean

In [5]:
x_bar = bmi_df["BMIPCT"].mean()
s = bmi_df["BMIPCT"].std()
n_bmi = len(bmi_df)

round(x_bar, 4), round(s, 4), n_bmi

(64.8207, 27.5168, 13062)

The sample mean BMI percentile is approximately 64.8207 based on a sample size of 13,062. This value is slightly below the benchmark value (μ0 = 65), suggesting a small difference that will be further tested in the inference stage.

### 3-2 Confidence Interval for Mean

In [6]:
t_crit = stats.t.ppf(0.975, df=n_bmi-1)
se_mean = s / np.sqrt(n_bmi)

ci_lower = x_bar - t_crit * se_mean
ci_upper = x_bar + t_crit * se_mean

(ci_lower, ci_upper)

(64.34874975076963, 65.29261659745646)

The 95% confidence interval for the mean BMI percentile is approximately (64.35, 65.29). Since the benchmark value (μ0 = 65) lies within this interval, there is no strong evidence that the true mean differs from 65.

### 3-3 One-Sample t-test

In [7]:
mu0 = 65

t_stat = (x_bar - mu0) / se_mean
p_value = 2 * (1 - stats.t.cdf(abs(t_stat), df=n_bmi-1))

t_stat, p_value

(-0.7447810972515212, 0.45641746445840203)

The t-test gives a p-value of approximately 0.4564, which is greater than the significance level (α = 0.05). Therefore, we fail to reject the null hypothesis. There is no sufficient evidence to conclude that the true mean BMI percentile is different from 65.

## 4. Final Interpretation

### Conclusion
For the proportion analysis, the sample proportion of students who feel sad or hopeless is approximately 0.30. The 95% confidence interval includes the benchmark value of 0.30, and the p-value from the one-sample proportion test is much greater than 0.05. Therefore, we fail to reject the null hypothesis and conclude that there is no sufficient evidence that the true proportion differs from 0.30.

For the mean analysis, the sample mean BMI percentile is approximately 64.82, which is slightly below the benchmark value of 65. The 95% confidence interval includes 65, and the p-value from the one-sample t-test is 0.4564, which is also greater than 0.05. Therefore, we fail to reject the null hypothesis and conclude that there is no sufficient evidence that the true mean BMI percentile differs from 65.

Overall, the inferential results are consistent with the patterns observed in the EDA. The proportion of students who feel sad or hopeless appears very close to the benchmark value, and the average BMI percentile is also close to 65. In addition, the extra EDA suggests a slight increase in the proportion of sadness across BMI groups, but this pattern should be interpreted cautiously because the current inference analysis did not directly test the association between BMI group and sadness.

Overall, the inferential results are consistent with the patterns observed in the EDA. The proportion of students who feel sad or hopeless appears very close to the benchmark value, and the average BMI percentile is also close to 65. In addition, the extra EDA suggests a slight increase in the proportion of sadness across BMI groups. To examine this relationship more formally, an additional chi-square test of independence is conducted below.

## 5. Additional Inferential Analysis: Association between BMI Group and SadOrHopeless
### Why a Different Method Is Needed

The first two analyses use one-sample inference because each question compares one sample statistic with a fixed benchmark value. Specifically, the proportion analysis compares the sample proportion with p0 = 0.30, and the mean analysis compares the sample mean with μ0 = 65.

However, the additional question is different. It does not compare one variable with a fixed benchmark. Instead, it examines whether two categorical variables, BMI group and SadOrHopeless status, are associated with each other.

Therefore, a chi-square test of independence is more appropriate. This method is used to test whether the distribution of sadness differs across BMI groups.
### 5-1 Create BMI Groups

In [8]:
joint_df["BMI_group"] = pd.qcut(
    joint_df["BMIPCT"],
    3,
    labels=["Low", "Medium", "High"]
)

joint_df[["BMIPCT", "BMI_group", "Sad_binary"]].head()

,BMIPCT,BMI_group,Sad_binary
0,66.531824,Medium,0
1,98.174319,High,1
2,33.075531,Low,0
3,45.688334,Low,0
4,62.390331,Medium,0


BMI percentile is divided into three groups (Low, Medium, High) based on quantiles.

This grouping ensures that each group contains a similar number of observations, allowing for a more balanced comparison of the proportion of students feeling sad or hopeless across BMI levels.

### 5-2 Contingency Table

In [9]:
contingency_table = pd.crosstab(joint_df["BMI_group"], joint_df["Sad_binary"])
contingency_table

Sad_binary,0,1
BMI_group,,
Low,3059,1242
Medium,3038,1263
High,2963,1337


The contingency table presents the observed frequencies of students who do and do not feel sad or hopeless within each BMI group.

From the table, we can observe that the number of students reporting sadness slightly increases from the Low to High BMI group. However, the differences across groups are relatively small, suggesting that any potential association may be weak.

These observed counts will be used in the chi-square test to formally evaluate whether the differences are statistically significant.

### 5-3 Chi-Square Test of Independence

In [10]:
chi2_stat, p_value_chi2, dof, expected = stats.chi2_contingency(contingency_table)

chi2_stat, p_value_chi2, dof

(5.576114451018698, 0.06154065747095707, 2)

The chi-square test examines whether BMI group and sadness status are independent.

The test statistic is approximately 5.576 with a p-value of 0.0615. Since the p-value is greater than the significance level of 0.05, we fail to reject the null hypothesis.

This indicates that there is no sufficient statistical evidence to conclude that BMI group and sadness status are associated.

### 5-4 Expected Counts

In [11]:
expected_df = pd.DataFrame(
    expected,
    index=contingency_table.index,
    columns=contingency_table.columns
)
expected_df

Sad_binary,0,1
BMI_group,,
Low,3020.234072,1280.765928
Medium,3020.234072,1280.765928
High,3019.531856,1280.468144


The expected counts represent the frequencies we would expect if BMI group and sadness status were independent.

All expected counts are sufficiently large (greater than 5), which satisfies the assumptions of the chi-square test. Therefore, the test results are reliable.

Comparing the observed and expected counts shows some differences across BMI groups, but these differences are not large enough to be statistically significant.

### 5-5 Conclusion for Additional Analysis

The p-value from the chi-square test is approximately 0.0615, which is greater than the significance level of 0.05. Therefore, we fail to reject the null hypothesis.

This suggests that there is not sufficient statistical evidence to conclude that BMI group and sadness status are associated.

Although the EDA showed a slight increasing trend in the proportion of sadness from the Low to High BMI group, this difference is not statistically significant at the 5% level.

This result indicates that the observed pattern may be due to random variation rather than a true underlying relationship.